typer/core.py

```diff
@@ -208,7 +208,19 @@ def _main(
             raise click.Abort() from e
         except KeyboardInterrupt as e:
             raise click.exceptions.Exit(130) from e
+        # TODO: When deprecating Click < 8.2 uncomment the next two lines
+        # except click.exceptions.NoArgsIsHelpError as e:
+        # raise click.exceptions.Exit(0) from e
         except click.ClickException as e:
+            # TODO: When deprecating Click < 8.2 remove this section [start]
+            _no_args_is_help_error = getattr(
```

Comment:
Rather than probing attrs and instance, I suggest using the exception we want to catch.
Click's __init__.py doesn't pull in the `NoArgsIsHelpError` exception into itself. So instead, we can easily do the same at the top of this file with;

```python
from click.exceptions import NoArgsIsHelpError
```

and then here simply the code to be

```python
...
except KeyboardInterrupt as e:
    raise click.exceptions.Exit(130) from e
except NoArgsIsHelpError as e:
    raise click.exceptions.Exit(0) from e
except click.ClickException as e:
```